src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIALIZATION_SIZE = 10;
+    private static final double ARRAY_RATIO = 1.5;
+    private T[] values;
+    private int currentSize;
+    private int lastElement = 0;
+
+    public ArrayList() {
+        currentSize = INITIALIZATION_SIZE;
+        values = (T[]) new Object[currentSize];
+    }
+
     @Override
     public void add(T value) {
-
+        if (lastElement >= currentSize) {
+            grow();
+        }
```

Comment:
move this check to method grow()
```suggestion
            grow();
```